File: src/main/java/com/xeoneux/caravansary/converter/RoomEntityToReservableRoomResponseConverter.java
package com.xeoneux.caravansary.converter;

import com.xeoneux.caravansary.entity.RoomEntity;
import com.xeoneux.caravansary.model.Links;
import com.xeoneux.caravansary.model.Self;
import com.xeoneux.caravansary.response.ReservableRoomResponse;
import com.xeoneux.caravansary.rest.ResourceConstants;
import org.springframework.core.convert.converter.Converter;

public class RoomEntityToReservableRoomResponseConverter
        implements Converter<RoomEntity, ReservableRoomResponse> {
    @Override
    public ReservableRoomResponse convert(RoomEntity source) {
        Self self = new Self();
        self.setRef(ResourceConstants.ROOM_RESERVATION_V1 + "/" + source.getId());

        Links links = new Links();
        links.setSelf(self);

        ReservableRoomResponse reservableRoomResponse = new ReservableRoomResponse();
        reservableRoomResponse.setId(source.getId());
        reservableRoomResponse.setRoomNumber(source.getRoomNumber());
        reservableRoomResponse.setPrice(Integer.valueOf(source.getPrice()));
        reservableRoomResponse.setLinks(links);

        return reservableRoomResponse;
    }
}

File: src/main/kotlin/com/xeoneux/caravansary/converter/RoomEntityToReservableRoomResponseConverter.kt
package com.xeoneux.caravansary.converter

import com.xeoneux.caravansary.entity.RoomEntity
import com.xeoneux.caravansary.model.Links
import com.xeoneux.caravansary.model.Self
import com.xeoneux.caravansary.response.ReservableRoomResponse
import com.xeoneux.caravansary.rest.ResourceConstants
import org.springframework.core.convert.converter.Converter

class RoomEntityToReservableRoomResponseConverter : Converter<RoomEntity, ReservableRoomResponse> {
    override fun convert(source: RoomEntity): ReservableRoomResponse {
        val self = Self()
        self.ref = ResourceConstants.ROOM_RESERVATION_V1 + "/" + source.id

        val links = Links()
        links.self = self

        val reservableRoomResponse = ReservableRoomResponse()
        reservableRoomResponse.id = source.id
        reservableRoomResponse.roomNumber = source.roomNumber
        reservableRoomResponse.price = Integer.valueOf(source.price!!)
        reservableRoomResponse.links = links

        return reservableRoomResponse
    }
}
